lib/resource-mapper.js

```diff
@@ -122,7 +122,7 @@ class ResourceMapper {
         if (!isIndex) {
           match = files.find(f => this._removeDollarExtension(f) === filename)
         // Check if the index file exists
-        } else if (files.includes(this._indexFilename)) {
+        } else if (files.includes(this._indexFilename) && contentType && contentType.includes(this._indexContentType)) {
```

Comment:
Hmm. First of all, this can't be correct; the last condition will be true with `text/htmlxyz` whereas we probably want it to fail. So why the choice for `includes` here? Is `contentType` expected to be an `Accept` header? That would not be a correct use of the parameter, which is intended for a single content type. (Namely in the case of a `PUT`, where the client sends a `Content-Type`.)

However, I don't understand the intuition behind this line. What is is supposed to mean? The index file can only be served if the user agent "sends" a content type, that just so happens to be the content type of the index file?

Reply:
Perhaps it is, because the contentType can have more values than „text/html” (IIRC a charset parameter or something). That's why an accept match isn't possible.
I had a „starts with” comparison in one of my projects, but a check against MIME types?
https://www.npmjs.com/package/@goa/content-type came after a short search for „RFC 7231”.

Reply:
I'd be surprised if a `GET` could be reduced to a single content type (it cannot), but indeed, content types can have arguments.